### PR TITLE
CMSCH #195 Refactor Community and Resort page

### LINKED ISSUE #195: refactor(ui): Community and resort page realignments
Root cause:

I guess the Community and Resort pages are kinda crammed and not too interesting.

Solution:

As seen in the image, on mobile the page is no so inviting. 

* The community logo should be aligned to right
* The title should be omitted (this requires a model change, maybe add an `isTitleOmittable` attribute to the Community model).
* *"A kör tevékenységei közé tartozik"* is too long, we do not really need this header, or rewrite it to *"Kulcsszavak"*. 
* There's too much space between the light-gray list items. We don't need that much, it just bugs the eye.
* Also, move the tags up under the light-gray list items. Have a bit more margin under the tags, so that there's distinctive spacing between the description and tags + light-gray list + logo.

Attachment:

Old look -> New look

image

### LINKED ISSUE #201: fix(ui): Unnecessarily large body text font
## Root cause:

https://github.com/kir-dev/cmsch/blob/08495b7bf5c96792ab8c3c5ed524c331856e5f71/src/main/client/src/utils/customTheme.ts#L36

Why tho? This strains my eyes and defeats good look&feel of any page and layout.

## Solution:

We should stay at the default Chakra font sizings. Comment out or delete this configuration object. Maybe differ from them ONLY on the home page where not much text is included.

## Attachment:

\-

### PR DESCRIPTION
**Pull request by triszt4n**
Closes #195 
Closes #201 if you allow it

## Attachments:

image

image

image

Tasty 🥪

### CODE REVIEW

File: src/main/client/src/components/@commons/DataSheet.tsx [hunk 1]
```diff
@@ -17,92 +16,118 @@ type DataSheetProps = {
...
+        <LinkButton href={organization.instagram} external leftIcon={<FaInstagram />} colorScheme="purple">
+          Instagram
+        </LinkButton>
+      )}
+    </Wrap>
+  </>
+)
+
+type DataFieldProps = {
+  icon: JSX.Element
```
Comment (triszt4n) [inline src/main/client/src/components/@commons/DataSheet.tsx:81]: Added icons lol

File: src/main/client/src/components/@commons/DataSheet.tsx [hunk 2]
```diff
@@ -17,92 +16,118 @@ type DataSheetProps = {
...
+const generateParagraphs = (paragraphs: string | string[]): JSX.Element => {
+  if (Array.isArray(paragraphs)) {
+    return (
+      <>
+        {paragraphs.map((paragraph, index) => (
+          <Paragraph key={`${index}@${paragraph.substring(0, 16)}`}>{paragraph}</Paragraph>
+        ))}
+      </>
+    )
+  }
```
Comment (triszt4n) [inline src/main/client/src/components/@commons/DataSheet.tsx:104]: Communities can add more paragraphs, not only one

File: src/main/client/src/content/communities.ts [hunk 3]
```diff
@@ -2,19 +2,32 @@ import { Community } from '../types/Organization'
 
 export const COMMUNITIES: Community[] = [
   {
-    id: 'kirdev',
+    id: 'kir-dev',
```
Comment (triszt4n) [inline src/main/client/src/content/communities.ts:5]: Let's stick to the dashed writing

File: src/main/client/src/content/communities.ts [hunk 4]
```diff
@@ -2,19 +2,32 @@ import { Community } from '../types/Organization'
...
     name: 'Kir-Dev',
+    hideName: true,
```
Comment (triszt4n) [inline src/main/client/src/content/communities.ts:7]: Can hide name, if logo is detailed enough as a title

File: src/main/client/src/content/communities.ts [hunk 5]
```diff
@@ -2,19 +2,32 @@ import { Community } from '../types/Organization'
...
+      “Speedrun munkacsoportunk” által került kivitelezésre nagyjából 2 hét alatt sok aktív tagunk nagyszerű erőbefektetésével. Hogy
+      betekintést nyerhessetek további izgalmas projektjeink – például a TanulóSCH, Schörpong, vagy az InduláSch – futásába, látogassatok
+      el weboldalunkra!`,
+      `Tanfolyamainkon megadjuk neked a szükséges kezdőlöketet, amivel Te is belevetheted magad a webfejlesztés izgalmas és kihívásokkal
+      teli világába. Lehetőséged van betekinteni az általunk használt Ruby on Rails, NodeJS és Spring Boot technológiáiba, a legkorszerűbb
+      programozási nyelvekbe. Kellemes atmoszféra, gyakori szakmai poénok és segítőkész csapattagok várnak.`,
+      `Ha érdeklődsz a kör iránt, akkor gyere el a tanfolyamainkra és gyűléseinkre!  A projektek vezetői szívesen bevezetnek mindenkit a
+      fejlesztés folyamatába. Keress minket a honlapunkon található elérhetőségeink egyikén!`
+    ],
+    interests: ['szoftver', 'webfejlesztés', 'programnyelvek'],
```
Comment (triszt4n) [inline src/main/client/src/content/communities.ts:26]: Sometimes less tags are more informative (3-4 is enough)

File: src/main/client/src/content/communities.ts [hunk 6]
```diff
@@ -2,19 +2,32 @@ import { Community } from '../types/Organization'
...
     website: 'https://kir-dev.sch.bme.hu',
-    facebook: 'https://kir-dev.sch.bme.hu',
-    instagram: 'https://kir-dev.sch.bme.hu',
-    application: 'https://kir-dev.sch.bme.hu',
+    facebook: 'https://fb.com/kirdevteam',
+    instagram: 'https://instagr.am/kir.dev',
+    application: 'https://kir-dev.sch.bme.hu/courses',
```
Comment (triszt4n) [inline src/main/client/src/content/communities.ts:30]: Defined these in advance

File: src/main/client/src/components/@commons/DataSheet.tsx [hunk 7]
```diff
@@ -17,92 +16,118 @@ type DataSheetProps = {
...
+            <Box>{organization.established}</Box>
+          </DataField>
         )}
-        {organization.application && (
-          <LinkButton href={organization.application} external leftIcon={<EditIcon />} colorScheme="brand">
-            Jelentkezés
-          </LinkButton>
+        {organization.email && (
+          <DataField icon={<FaAt />} label="E-mail">
+            <Box>{organization.email.replace('@', '[at]')}</Box>
```
Comment (triszt4n) [inline src/main/client/src/components/@commons/DataSheet.tsx:31]: Replacing the at sign for security purposes

File: src/main/client/src/components/@commons/DataSheet.tsx [hunk 8]
```diff
@@ -17,92 +16,118 @@ type DataSheetProps = {
...
         )}
-      </Wrap>
-    </>
-  )
+      </VStack>
+      {generateLogo(organization)}
+    </Flex>
+    {organization.descriptionParagraphs && generateParagraphs(organization.descriptionParagraphs)}
+
+    <Wrap marginTop={10} justify={{ base: 'center', md: 'flex-end' }}>
```
Comment (triszt4n) [inline src/main/client/src/components/@commons/DataSheet.tsx:55]: Replaced the `justify={['center', 'center', 'flex-start']}` here to `justify={{ base: 'center', md: 'flex-end'`. Reason? Using objects with named breakpoints like `base` and `md` are more informative than an array of values.

File: src/main/client/src/components/@commons/DataSheet.tsx [hunk 9]
```diff
@@ -17,92 +16,118 @@ type DataSheetProps = {
...
+type DataFieldProps = {
+  icon: JSX.Element
+  label: string
 }
 
-const DataGrid: React.FC = ({ children }) => {
+const DataField: React.FC<DataFieldProps> = ({ icon, label, children }) => {
   return (
-    <Grid templateColumns="repeat(2,auto)" mt={5} gap={5} width="fit-content" mr={5}>
+    <HStack color={useColorModeValue('gray.700', 'gray.200')}>
```
Comment (triszt4n) [inline src/main/client/src/components/@commons/DataSheet.tsx:87]: Might change for more ligher grays here

File: src/main/client/src/components/@commons/DataSheet.tsx [hunk 10]
```diff
@@ -17,92 +16,118 @@ type DataSheetProps = {
...
+      <Box fontWeight={700}>{label}</Box>
       {children}
-    </Grid>
+    </HStack>
   )
 }
 
-type DataFieldProps = {
-  label: string
+const generateParagraphs = (paragraphs: string | string[]): JSX.Element => {
```
Comment (triszt4n) [inline src/main/client/src/components/@commons/DataSheet.tsx:95]: It's a good practise to lift out code parts which only generate some elements.

File: src/main/client/src/components/@commons/DataSheet.tsx [hunk 11]
```diff
@@ -17,92 +16,118 @@ type DataSheetProps = {
...
+  if (Array.isArray(paragraphs)) {
+    return (
+      <>
+        {paragraphs.map((paragraph, index) => (
+          <Paragraph key={`${index}@${paragraph.substring(0, 16)}`}>{paragraph}</Paragraph>
```
Comment (triszt4n) [inline src/main/client/src/components/@commons/DataSheet.tsx:100]: Never left out `key` prop on parent components of `map` or `forEach` functions!

File: src/main/client/src/components/@commons/DataSheet.tsx [hunk 12]
```diff
@@ -17,92 +16,118 @@ type DataSheetProps = {
...
+        my={2}
+        alignSelf={{ base: 'flex-end', sm: 'flex-start' }}
+        src={logoSource}
+        alt={org.name}
+        maxH={{ base: '10rem', sm: '10rem', md: '16rem' }}
+        maxW={{ base: '16rem', sm: '10rem', md: '16rem' }}
+        objectFit="contain"
+      />
+    )
+  }
```
Comment (triszt4n) [inline src/main/client/src/components/@commons/DataSheet.tsx:130]: Hook-independent code parts can be lifted out into functions to spare the eye strain while reading a React Function Components structure.

File: src/main/client/src/utils/customTheme.ts [hunk 13]
```diff
@@ -33,7 +33,7 @@ const customTheme = extendTheme({
     },
     Text: {
       baseStyle: {
-        fontSize: 'lg'
+        // fontSize: 'lg'
```
Comment (triszt4n) [inline src/main/client/src/utils/customTheme.ts:36]: On mobile, the fonts are too large, very un-readable. See #201

**Review by Tschonti — APPROVED** [review]
I'd prefer if the social buttons were on the left, but that might just be me.

**Review by DannySS123 — APPROVED** [review]
Nice work!  🎉 
The only thing that looks a bit strange to me is the size of the Simonyi logo. Maybe we could set the maxH (perhaps of all logos) to 10rem, that lookes better IMO.

Now:
image

With maxH = 10rem:
image

**Review by berenteb — APPROVED** [review]
You may have gone through a lot of concept, I accept this solution, looks great!

**Pull request merged**